api/v1alpha1/controllermanagerconfig_types.go

```diff
@@ -105,6 +105,16 @@ type ControllerManagerConfigurationSpec struct {
 	// Webhook contains the controllers webhook configuration
 	// +optional
 	Webhook ControllerWebhook `json:"webhook,omitempty"`
+
+	// ClassName contains the name of a class to watch CRs for. Others will be ignored.
+	// If unset all will be watched.
+	// +optional
+	ClassName string `json:"className,omitempty"`
+
+	// If MissingClassName is set and ClassName is set, any CR without a ClassName
```

Comment:
Is the presence of and naming of this configurable conventional? What's the use case?

Reply:
Not sure how much precedent there is. I can compare against what they did in ingress-nginx.

As for the use case, say you deploy one spire instance. no class name on it. users upload crds with no class name attached.

Then a second spire instance is desired. So, how is it safe to have both? A migration procedure is needed.

First, the existing spire server is updated to have
ClassName: foo
MissingClassName: true

So it will now have a name, and it will still handle all existing CR's without a ClassName.

The new spire server can then be installed with
ClassName: bar
MissingClassName: false

and new CRs of ClassName bar can be added to the cluster.

Reply:
Is the problem here backwards compatibility? If starting from scratch we would probably have `ClassName` set to a default value and then the separate "watch without class" bool set to false by default.

Reply:
I see. How about `--watch-classless`?

In terms of defaults, we need to preserve the default behavior of "handle everything", which should be the behavior if ClassName is empty. Using WatchClassless to allow a controller to simultaneously handle CR's without a ClassName as well as CRs matching the class name seems like a fine migration behavior.

Reply:
Yeah, the default is to watch all. (className empty)

Reply:
Will rename the missing flag.

Reply:
Done